src/shims/io.rs

```diff
@@ -42,16 +42,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let flag = this.read_scalar(flag_op)?.to_i32()?;
 
-        if flag != this.eval_libc_i32("O_RDONLY")? && flag != this.eval_libc_i32("O_CLOEXEC")? {
```

Comment:
This seems to entirely remove the check for unknown flags, and instead silently ignore them. That's not great.

Also, the `O_CLOEXEC` vs `FD_CLOEXEC` part is confusing (not in your implementation but in Linux/POSIX^^). Is it the case that these flags have the same value?

Reply:
> This seems to entirely remove the check for unknown flags, and instead silently ignore them. That's not great.

I think i can add this check by setting the correct bits to zero for each supported flag and then checking that the flag is exactly zero. Meaning that no other option was used (unless having a zero somewhere enables an option like with read only mode ¬¬)

> Also, the `O_CLOEXEC` vs `FD_CLOEXEC` part is confusing (not in your implementation but in Linux/POSIX^^). Is it the case that these flags have the same value?

I checked libc and they are different, not only that, but `FD_CLOEXEC == O_WRONLY`.

Reply:
See https://github.com/rust-lang/miri/pull/962#discussion_r333901667

> I think i can add this check by setting the correct bits to zero for each supported flag and then checking that the flag is exactly zero. Meaning that no other option was used (unless having a zero somewhere enables an option like with read only mode ¬¬)

Something like that should work, yes.

Reply:
I got that. ;) And yes.

Reply:
https://github.com/rust-lang/miri/pull/987 takes care of this (hopefully)